src/decode.rs

```diff
@@ -4364,20 +4361,10 @@ pub(crate) unsafe fn rav1d_decode_frame_init(
         }
 
         let cf_sz = (num_sb128 * size_mul[0] as c_int) << hbd;
-        if cf_sz != f.frame_thread.cf_sz {
-            rav1d_freep_aligned(&mut f.frame_thread.cf as *mut *mut DynCoef as *mut c_void);
-            f.frame_thread.cf =
-                rav1d_alloc_aligned(cf_sz as usize * 128 * 128 / 2, 64) as *mut DynCoef;
-            if f.frame_thread.cf.is_null() {
-                f.frame_thread.cf_sz = 0;
-                return Err(ENOMEM);
-            }
-            slice::from_raw_parts_mut(
-                f.frame_thread.cf.cast::<u8>(),
-                usize::try_from(cf_sz).unwrap() * 128 * 128 / 2,
-            )
-            .fill(0);
-            f.frame_thread.cf_sz = cf_sz;
+        if cf_sz as usize != f.frame_thread.cf_sz() {
+            f.frame_thread.cf.clear();
+            // TODO: Fallible allocation
+            f.frame_thread.cf.resize(cf_sz as usize * 128 * 128 / 2, 0);
         }
```

Comment:
```suggestion
        // TODO: Fallible allocation
        f.frame_thread.cf.resize(cf_sz as usize * 128 * 128 / 2, 0);
```

Reply:
This doesn't have the same semantics as the original C. The original logic was filling the entire buffer with 0 after resizing, not just the new elements. Are you sure we can change the semantics here without causing issues? I can try it and at least see if it causes any test failures.

Reply:
Tests seem to pass so I've made the change.

Reply:
Hmm, I *think* it's just because C made an entirely new allocation so they have to, but I'm not sure.  Though if the tests pass it's probably okay?